src/clojure_lsp/handlers.clj

```diff
@@ -215,61 +215,73 @@
                              [(:sym usage) usages])
                            (into {}))]
     (when cursor-value
-      (concat
-        (->> local-env
-             (filter (comp :declare :tags))
+      (if (contains? (:tags cursor-usage) :refer)
+        ;; If the cursor is within a :refer then the sole
+        ;; completions should be symbols within the namespace
+        ;; that's being referred
+        (->> (get remotes-by-ns (:ns cursor-usage))
+             (filter #(every? (:tags %) [:declare :public]))
```

Comment:
I get a null pointer exception here when `:tags` are nil.

Reply:
Ah, in which cases is `:tags` nil? (So I can add tests for it too)